### PR TITLE
Info dump 44-53/66: Lots of stuff

### PR DESCRIPTION
**Pull request by End45**
Adds functions mostly related to:

- Overworld map markers
- Clear marks (shown when loading a save file)
- Item spawn lists
- Monster initialization
- Ranks
- Bag and items
- Weather graphical effects
- Picking up and throwing items

`SpawnEnemyItemDrop` has been renamed to `PlaceThrownItem`. I have no idea why it had that name originally, if someone knows what that was supposed to mean, let me know.

### CODE REVIEW

**Comment by Jawshoeuh** [comment]
I don't know why it was originally named that (perhaps a previous assumption by someone because TrySpawnEnemyItemDrop calls it through the wrapper?); however, I don't know if PlaceThrownItem is accurate either because the function is used to drop coins for Pay Day and the held item for Knock Off. I think the usage is more general.

**Comment by End45** [comment]
> I don't know why it was originally named that (perhaps a previous assumption by someone because TrySpawnEnemyItemDrop calls it through the wrapper?); however, I don't know if PlaceThrownItem is accurate either because the function is used to drop coins for Pay Day and the held item for Knock Off. I think the usage is more general.

So it's used when defeated enemies drop stuff, after a thrown item hits a wall and drops to the floor, for Pay Day and for Knock Off. But not when an item is manually placed.
Any ideas for a better name? Maybe `PlaceDroppedItem`?

**Comment by UsernameFodder** [comment]
> > I don't know why it was originally named that (perhaps a previous assumption by someone because TrySpawnEnemyItemDrop calls it through the wrapper?); however, I don't know if PlaceThrownItem is accurate either because the function is used to drop coins for Pay Day and the held item for Knock Off. I think the usage is more general.
> 
> So it's used when defeated enemies drop stuff, after a thrown item hits a wall and drops to the floor, for Pay Day and for Knock Off. But not when an item is manually placed. Any ideas for a better name? Maybe `PlaceDroppedItem`?

I apparently added this function in https://github.com/UsernameFodder/pmdsky-debug/pull/91. The usage I found was for spawning certain special item drops like treasure boxes and Unown stones, but I probably didn't realize it was more general.

Can we call this `SpawnDroppedItem`? It ultimately ends up calling `SpawnItem` after doing stuff specific to dropped items.

File: symbols/overlay29.yml [hunk 1]
```diff
@@ -6797,28 +6940,41 @@ overlay29:
...
       description: |-
-        Wraps SpawnEnemyItemDrop in a more convenient interface.
+        Wraps PlaceThrownItem in a more convenient interface.
         
         r0: entity
         r1: position
         r2: item
         r3: ?
-    - name: SpawnEnemyItemDrop
+    - name: PlaceThrownItem
```
Comment (UsernameFodder) [inline symbols/overlay29.yml:6967]: Seems to need renaming per discussion

File: symbols/overlay29.yml [hunk 2]
```diff
@@ -1467,6 +1519,33 @@ overlay29:
...
+    - name: UseThrowableItem
+      address:
+        EU: 0x22F5E78
+        NA: 0x22F54BC
+      description: |-
+        Makes a monster use the specified throwable item.
+        
+        If the item's category is CATEGORY_THROWN_LINE or CATEGORY_THROWN_ARC, the game will attempt to decrement the count of the used item by 1. If it's not or there's only 1 item left, it is destroyed instead.
+        
+        r0: User (monster who used the item)
```
Comment (UsernameFodder) [inline symbols/overlay29.yml:1548]: With these three functions, Is the actual item to be used stored on the user struct somewhere? Could use clarification what "specified" means.

**Review by UsernameFodder — CHANGES_REQUESTED** [review]
(I made a smattering of small tweaks)

File: headers/types/ground_mode/ground_mode.h [hunk 3]
```diff
@@ -374,4 +374,12 @@ struct ground_weather_entry {
 };
 ASSERT_SIZE(struct ground_weather_entry, 4);
 
+// Used to store the coordinates of a position on the World Map. The values are shifted 8 bits to
+// the left, so they are probably fixed-point decimals instead of integers.
+struct map_pos {
```
[the remark this answers is not in the document]
Comment (End45) [reply headers/types/ground_mode/ground_mode.h:379]: I didn't know that type was a thing. I'll change it.

File: headers/types/ground_mode/ground_mode.h [hunk 4]
```diff
@@ -374,4 +374,12 @@ struct ground_weather_entry {
 };
 ASSERT_SIZE(struct ground_weather_entry, 4);
 
+// Used to store the coordinates of a position on the World Map. The values are shifted 8 bits to
+// the left, so they are probably fixed-point decimals instead of integers.
```
[the remark this answers is not in the document]
Comment (End45) [reply headers/types/ground_mode/ground_mode.h:378]: Makes sense

File: symbols/arm9.yml [hunk 5]
```diff
@@ -3845,6 +3856,17 @@ arm9:
         Load and initialise the alert sprite, storing the result in ALERT_ANIMATION_CONTROL
         
         No params.
+    - name: PrintClearMark
+      address:
+        EU: 0x202A6D8
+        NA: 0x202A3E4
+      description: |-
+        Prints the specified clear mark on the screen.
```
[the remark this answers is not in the document]
Comment (End45) [reply symbols/arm9.yml:3864]: Clear marks are those icons with the faces of the bosses that show up when you resume a saved game. They are stored in FONT/clrmark1.w16, so I believe that's their internal name.
But I'll clarify this a bit.

File: symbols/overlay29.yml [hunk 6]
```diff
@@ -631,11 +651,23 @@ overlay29:
...
         
         r0: quick_saved
         r1: ???
         r2: special_process
+    - name: GetItemIdToSpawn
+      address:
+        EU: 0x22E8474
+        NA: 0x22E7AC4
+      description: |-
+        Get the id of the item to be randomly spawned.
```
[the remark this answers is not in the document]
Comment (End45) [reply symbols/overlay29.yml:666]: I copied that from `GetMonsterIdToSpawn`. Both are generic utilities to pick a random entry from a spawn lisst.
Should we change both?

File: symbols/overlay29.yml (hunk 1 above)
Comment (End45) [reply symbols/overlay29.yml:6967]: Renamed to `SpawnDroppedItem` as suggested.

File: symbols/overlay29.yml (hunk 2 above)
Comment (End45) [reply symbols/overlay29.yml]: I'll clarify it.

File: symbols/overlay29.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (UsernameFodder) [reply symbols/overlay29.yml:666]: I see. Yeah let's change both to be clearer.

File: symbols/overlay29.yml (hunk 1 above)
Comment (UsernameFodder) [reply symbols/overlay29.yml]: I tweaked the description a bit more, but otherwise this looks good.

**Review by UsernameFodder — APPROVED** [review]
Can't believe we're past 50 already!

**Pull request merged**